### PR TITLE
Feature/quadratic problem scale

### PR DESCRIPTION
**Pull request by michalk8**
In this PR:
- add `scale_cost` to `QuadraticProblem` which updates the `scale_cost` in all geometries
  when passed as `False` (default), it keeps the orignal `scale_cost` in the geometries
- allow `scale_cost` in geometries to be `bool`: if `True` a default is used (`mean` for `LRGeometry` and `PointCloud`, `max_cost` for `Geometry`, but am not very opinionated here)
- update CI to run on 3.8 (numpy support for 3.7 ended last December) and use only 2 workers (win/linux machines have 2 CPUs only, see [here](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners))
- fix some warnings in tests
- increase tolerance in `test_apply_squared` from `1e-4` to `5e-4` (was locally failing, also on `main`)

### CODE REVIEW

File: ott/core/quad_problems.py [hunk 1]
```diff
@@ -106,6 +107,10 @@ def __init__(self,
         i.e. problem = purely quadratic + fused_penalty * linear problem. If
         fused_penalty is None but geom_xy is passed, fused_penalty is set by
         default to 1.0, equal to 0.0 otherwise.
+      scale_cost
+        option to rescale the cost matrices. If `True`, use the default for
+        each geometry. If `False`, keep the original scaling in geometries.
+        If `None`, do not scale the cost matrices.
```
Comment (marcocuturi) [inline ott/core/quad_problems.py:113]: mention when 'str' is passed

File: ott/core/quad_problems.py [hunk 2]
```diff
@@ -493,6 +498,17 @@ def update_epsilon_unbalanced(epsilon, transport_mass):
   return updated_epsilon
 
 
+def update_geom_scale_cost(geom: Optional[geometry.Geometry],
```
Comment (marcocuturi) [inline ott/core/quad_problems.py:501]: maybe this should be in geometry.py and take self as first argument?

File: ott/geometry/geometry.py [hunk 3]
```diff
@@ -73,14 +73,15 @@ def __init__(self,
...
+        If `True`, use 'max_cost'.
       **kwargs: additional kwargs to epsilon.
     """
     self._cost_matrix = cost_matrix
     self._kernel_matrix = kernel_matrix
     self._epsilon_init = epsilon
     self._relative_epsilon = relative_epsilon
     self._scale_epsilon = scale_epsilon
-    self._scale_cost = scale_cost
+    self._scale_cost = "max_cost" if scale_cost is True else scale_cost
```
Comment (marcocuturi) [inline ott/geometry/geometry.py:84]: would rather use mean for consistency.

File: tests/core/gromov_wasserstein_test.py [hunk 4]
```diff
@@ -231,6 +233,25 @@ def test_gw_lr_fused(self):
...
+    epsilon = 0.1
+    fused_penalty = 1
+    geom_x = pointcloud.PointCloud(self.x, scale_cost=None)
+    geom_y = pointcloud.PointCloud(self.y, scale_cost=None)
+    geom_xy = pointcloud.PointCloud(self.xx, self.yy, scale_cost=None)
+    geom_x_scaled = pointcloud.PointCloud(self.x, scale_cost=scale_cost)
+    geom_y_scaled = pointcloud.PointCloud(self.y, scale_cost=scale_cost)
+    geom_xy_scaled = pointcloud.PointCloud(self.xx, self.yy, scale_cost=scale_cost)
+
+    gt = gromov_wasserstein.gromov_wasserstein(geom_xx=geom_x_scaled, geom_yy=geom_y_scaled, geom_xy=geom_xy_scaled,
```
Comment (marcocuturi) [inline tests/core/gromov_wasserstein_test.py:247]: 80 chars (here and elsewhere)

File: tests/tools/sinkhorn_divergence_test.py [hunk 5]
```diff
@@ -187,8 +187,8 @@ def test_segment_sinkhorn_result(self, shuffle):
     if shuffle:
       # Now, shuffle the order of both arrays, but
       # still maintain the segment assignments:
-      idx_x = jax.random.shuffle(rngs[2], jnp.arange(x.shape[0] * 2))
-      idx_y = jax.random.shuffle(rngs[3], jnp.arange(y.shape[0] * 2))
+      idx_x = jax.random.permutation(rngs[2], jnp.arange(x.shape[0] * 2), independent=True)
```
Comment (marcocuturi) [inline tests/tools/sinkhorn_divergence_test.py:190]: thanks!

**Review by marcocuturi — COMMENTED** [review]
Thanks Michal!

File: ott/geometry/geometry.py (hunk 3 above)
Comment (michalk8) [reply ott/geometry/geometry.py]: Done.

File: tests/core/gromov_wasserstein_test.py (hunk 4 above)
Comment (michalk8) [reply tests/core/gromov_wasserstein_test.py]: Done, although in #43 I will argue to go for `120`.

File: ott/core/quad_problems.py (hunk 1 above)
Comment (michalk8) [reply ott/core/quad_problems.py]: Done.

File: ott/core/quad_problems.py (hunk 2 above)
Comment (michalk8) [reply ott/core/quad_problems.py]: Originally, I was against this (as the method either returns unmodified `self` or a new instance), but I changed my mind and think your suggestion is better.
Done as a private method `_set_scale_cost(self, scale_cost)`.

**Comment by michalk8** [comment]
When updating the docstrings for `scale_cost`, I noticed there are many warnings with the docs (many lists/equations were incorrectly rendered), so I fixed it as well. Some warnings are still left, though they are not related to syntax.
Furthermore, some link to classes were broken (e.g., Linear/QuadradProblem), fixed whatever I could. Biggest improvement can be seen in the docs for `sinkhorn` function.

I also noticed that people can't click on, the functions from the classes, e.g. [PointCloud](https://ott-jax.readthedocs.io/en/latest/_autosummary/ott.geometry.pointcloud.PointCloud.html#ott.geometry.pointcloud.PointCloud).
I've added a very simple template which fixes it. Overall, docs will need a proper fix, this is just a triage (e.g. the package versions differ in multiple places).

Lastly, some tests are still failing, but they are afaik unrelated to this PR. I think the tolerances are too low, will check later (locally all pass).

**Comment by marcocuturi** [comment]
I think the tests are a bit flaky indeed. On github this seems to be a timeout issue.

Thanks a ton for this PR Michal, this is really great!

**Pull request merged**